core/src/main/java/org/pentaho/platform/scheduler2/action/SchedulerOutputPathResolver.java

```diff
@@ -116,42 +117,53 @@ public void setActionUser( String actionUser ) {
   @Override
   public String resolveOutputFilePath() throws SchedulerException {
 
-    final String fileNamePattern = getFilename();
-    final String outputFilePath = getDirectory();
-
+    // IMPROVEMENT: This does not belong here. Move to the caller.
     boolean scheduleAllowed = isScheduleAllowed();
     if ( !scheduleAllowed ) {
       throw new SchedulerException( Messages.getInstance().getString(
-        "QuartzScheduler.ERROR_0009_SCHEDULING_IS_NOT_ALLOWED_AFTER_CHANGE", getJobName(), this.actionUser ) );
+        "QuartzScheduler.ERROR_0009_SCHEDULING_IS_NOT_ALLOWED_AFTER_CHANGE", getJobName(), getActionUser() ) );
     }
 
     // Enclose validation logic in the context of the job creator's session, not the current session
-    final Callable<String> callable = new Callable<String>() {
```

Comment:
Moved this nested callable to its own class method and replaced by a method reference.